doc/guides/modules.rst

```diff
@@ -487,6 +486,100 @@ variables. For example:
 
   include(${ZEPHYR_CURRENT_MODULE_DIR}/cmake/code.cmake)
 
+.. _modules_module_root:
+
+Module root
+===========
+
+The Zephyr repository contain :file:`CMakeLists.txt` and :file:`Kconfig` build
+files for certain known Zephyr modules.
+
+Those build files are located under
+
+.. code-block:: none
+
+   <ZEPHYR_BASE>
+   └── modules
+       └── <module_name>
+           ├── CMakeLists.txt
+           └── Kconfig.<module_name>
+
+You can create a similar ``MODULE_ROOT`` for additional modules, and make those
+modules known to Zephyr.
+
+Create a ``MODULE_ROOT`` with the following structure
+
+.. code-block:: none
+
+   <MODULE_ROOT>
+   └── modules
+       ├── modules.cmake
+       └── <module_name>
+           ├── CMakeLists.txt
+           └── Kconfig.<module_name>
+
+and then build your application by specifying ``-DMODULE_ROOT`` parameter to the
+CMake build system. The ``MODULE_ROOT`` accepts a CMake list of roots as
+argument.
+
+.. note::
+
+   ``ZEPHYR_BASE`` is always added as a ``MODULE_ROOT`` with the lowest priority.
+   This allows you to overrule any build files under
+   ``<ZEPHYR_BASE>/modules/<module_name>`` with your own implementation your own
+   ``MODULE_ROOT``.
+
+The :file:`modules.cmake` file must contain the logic that specifies the build
+files for Zephyr modules via specifically named CMake variables.
+
+To include a module's CMake file, the variable ``ZEPHYR_<MODULE_NAME>_CMAKE_DIR``
+is used.
+
+To include a module's Kconfig file, the variable ``ZEPHYR_<MODULE_NAME>_KCONFIG``
+is used.
+
+The following is an example on how to add support the the ``FOO`` module.
+
+Create the following structure
+
+.. code-block:: none
+
+   <MODULE_ROOT>
+   └── modules
+       ├── modules.cmake
+       └── foo
+           ├── CMakeLists.txt
+           └── Kconfig.foo
+
+and inside the :file:`modules.cmake` file, add the following content
+
+.. code-block:: cmake
+
+   set(ZEPHYR_FOO_CMAKE_DIR ${CMAKE_CURRENT_LIST_DIR}/foo)
+   set(ZEPHYR_FOO_KCONFIG   ${CMAKE_CURRENT_LIST_DIR}/foo/Kconfig.foo)
+
+.. _modules_build_files_extern:
+
+Build files in a module root
```

Comment:
The title reads a bit wonky since it's unclear whether "build" is a verb or adjective. I'm not sure this even needs to be a separate section since it's so related to the previous.

Reply:
>  I'm not sure this even needs to be a separate section since it's so related to the previous.

If you look at the menu, it should have it's own section:
![image](https://user-images.githubusercontent.com/9802583/103763028-8888c000-5019-11eb-9ec5-34a884f59a3e.png)

That will make it visible that there are two sections:
- `Build files` 
- `Build files in module root`
So depending on what you need to do, you can go to the right section, otherwise a user might just go to the `Build files` section  when they actually needed the `Build files in module root`.

Reply:
Ok
```suggestion
Build files placed in a module root
```

Reply:
Choosing the first version, as `Build files` is the important part of the info the reader need to notice.